src/appimagetool_fetch_runtime.cpp

```diff
@@ -57,8 +108,87 @@ class GetRequest {
         }
     }
 
+    /**
+     * Query data from CURL* handle.
+     * @tparam T name of the type curl_easy_getinfo will return for the given option
+     * @param option option to query
+     * @return value returned by libcurl
+     */
+    template<typename T>
+    T getOption(CURLINFO option) {
+        T temp;
+        checkForCurlError(curl_easy_getinfo(_handle, option, &temp));
+        return temp;
+    }
+
+    /**
+     * Set options within CURL* handle.
+     * @tparam T name of the input variable's type (usually deduced automatically, you do not have to pass this
+     *     explicitly)
+     * @param option option to set using curl_easy_setinfo
+     * @param value value to set
+     */
+    template<typename T>
+    void setOption(CURLoption option, T value) {
+        checkForCurlError(curl_easy_setopt(_handle, option, value));
+    }
+
+    void setUpTlsCaChainCompatibility(bool verbose) {
+        // from curl 7.84.0 on, one can query the default values and check if these files or directories exist
+        // if not, we anyway run the detection
+#define querying_supported LIBCURL_VERSION_NUM >= CURL_VERSION_BITS(7, 84, 0)
+#if querying_supported
+        {
+            const auto caInfo = getOption<char*>(CURLINFO_CAINFO);
+            if (std::filesystem::exists(caInfo)) {
+                if (verbose) {
+                    std::cerr << "libcurl's default CA certificate bundle file " << caInfo << " was found on this system" << std::endl;
+                }
+                return;
+            }
+        }
+
+        {
+            const auto caPath = getOption<char*>(CURLINFO_CAPATH);
+            if (std::filesystem::is_directory(caPath) {
+                if (verbose) {
+                    std::cerr << "libcurl's default CA certificate bundle directory " << caPath
+                              << " was found on this system" << std::endl;
+                }
+                return;
+            }
+        }
+#else
+#warning "libcurl version too old, not trying to use default values for system-provided CA certificate bundles"
```

Comment:
What does this actually mean then? Is that just for info, slighly annoying, or will result in broken functionality?

Reply:
Using libcurl's default may provide us with another location we do not (yet) support. We do not _strictly_ need this, though. And we know the warning won't be shown in our Alpine build environment because libcurl is recent enough. However, during development (in my case, I'm using some Ubuntu flavor), I cannot just use it. Instead of just deactivating the code, I want to explicitly tell people in such environments that this minor(!) feature is not available.

Therefore, the message itself is purely informational, the preprocessor code is needed anyway.